drools-game-engine-horserace-model/src/main/java/org/drools/game/horserace/model/Checkpoint.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * To change this license header, choose License Headers in Project Properties.
+ * To change this template file, choose Tools | Templates
+ * and open the template in the editor.
+ */
+package org.drools.game.horserace.model;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ *
+ * @author Samuel
+ */
+public class Checkpoint {
+
+    private String id;
+    private Checkpoint required;
+    private List<String> playersCrossedCheckpoint = new ArrayList<>();
```

Comment:
This is a little bit odd, the checkpoint shouldn't have to retain which players went through it. 
We can have a rule that creates a new type (using declare in the rules) to represent that information. By using those types, we can also avoid the "Checkpoint required" as well. Does it make sense?